### DISCUSSION TITLE
Sitemap generated with relative URLs only

### DISCUSSION THREAD

**Discussion by ISSOtm** [community/community#23341] [Other Feature Feedback, Questions, & Ideas]
<p>The problem was raised in <a href="https://github.com/gbdev/rgbds-www/issues/25" rel="noopener nofollow ugc">this issue</a>, where <a href="https://rgbds.gbdev.io/sitemap.xml" rel="noopener nofollow ugc">our sitemap</a> only contains relative URLs, which are disallowed by the spec and rejected by at least Google.</p>
<p>The sitemap is generated by the <code>jekyll-sitemap</code> plugin, and I just updated it to the latest version allowed by Pages, so that isn’t the issue.</p>
<p>What’s weird is that a local build produces a correct sitemap (<code>http://localhost:4000/docs/</code> etc.), only the one auto-built by Pages has this issue—thus, I believe that the way Pages overrides the <code>url</code> and/or <code>baseurl</code> config attrs breaks this plugin.</p>
<p>Are we missing something here? Thanks in advance!</p>

**Comment by yoannchaudet** [comment]
<p>Hello!</p>
<p>Jekyll (or the <code>jekyll-sitemap</code> plugin here) does not know the absolute URL of your website unless you specify it in your configuration file.</p>
<p>Have you tried setting the <code>url</code> to <code>https://rgbds.gbdev.io</code> in your <a href="https://github.com/gbdev/rgbds-www/blob/b90a8504da015f26308714286e21c221c246a71f/_config.yml#L17">configuration file</a> already?</p>

**Comment by ISSOtm** [comment]
<p>Yes, <a href="https://github.com/gbdev/rgbds-www/commit/26781e06a486211cf086d8266be2bfb144cf2af7" rel="noopener nofollow ugc">I have</a>, and even with a cache-less refresh (Ctrl+F5), the URLs in the sitemap were still relative. That’s why I’m assuming it must be something with the way GH Pages overrides this setting. Or have I missed anything?</p>

**Comment by yoannchaudet** [comment]
<p>I confirmed the <code>jekyll-sitemap</code> plugin can handle absolute paths just fine.</p>
<p>I took you repo and <a href="https://github.com/yoannchaudet/rgbds-www/commit/b2cb1e495d5a68cb04d411d1ccab5b275e5799ce">re-applied your configuration changes</a>. The sitemap looks great: <a href="https://yoannchaudet.github.io/rgbds-www/sitemap.xml">https://yoannchaudet.github.io/rgbds-www/sitemap.xml</a>.</p>
<p>Pages assets are served behind a CDN so you may have to wait up to 10 minutes for full propagation. A Ctrl+F5 refresh would not make our CDN propagate faster :slight_smile:</p>
<p>I suggest you re-apply the configuration change and give it a bit of time after a Pages rebuild for the changes to be reflected.</p>

**Comment by ISSOtm** [comment]
<p>Alright, <a href="https://github.com/gbdev/rgbds-www/commit/0c9b420bc7c0d71bfcf57f97281c14c9288c9607" rel="noopener nofollow ugc">I’ve tried again</a>, I’ll get back in an hour or so ^^</p>
<p>[EDIT] Well, looks like it didn’t take 10 minutes for me to see the change this time. Thanks!</p>
